crypto/x509/pcy_map.c

```diff
@@ -73,10 +73,8 @@ int ossl_policy_cache_set_mapping(X509 *x, POLICY_MAPPINGS *maps)
 
     ret = 1;
  bad_mapping:
-    if (ret == -1 && CRYPTO_THREAD_write_lock(x->lock)) {
+    if (ret == -1)
         x->ex_flags |= EXFLAG_INVALID_POLICY;
```

Comment:
You could even remove this completely since `policy_cache_new` does it again anyway.

Reply:
yeah, that would remove the coverity as well.